persistence-modules/hibernate-mapping/src/main/java/com/baeldung/hibernate/pojo/inheritance/Laptop.java

```diff
@@ -0,0 +1,39 @@
+package com.baeldung.hibernate.pojo.inheritance;
+
+import org.hibernate.annotations.Polymorphism;
+import org.hibernate.annotations.PolymorphismType;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+
+@Entity
+@Polymorphism(type = PolymorphismType.IMPLICIT)
+public class Laptop implements Item {
```

Comment:
Do we need this new file?

Reply:
I added this file because when I query by an interface if that interface doesn't have a class that implements that interface and have this annotation: `@Polymorphism(type = PolymorphismType.IMPLICIT)` it will throw an error. This way the interface Item was implemented in two classes: Laptop and Bag. The class Bag has the annotation Polymorphism with explicit annotation the other one (Laptop )has IMPLICIT Annotation.